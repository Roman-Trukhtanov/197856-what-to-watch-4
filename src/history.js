import {createBrowserHistory} from "history";
import {AppRoute, HistoryAction} from "./const";

const history = createBrowserHistory();

export const historyGoBack = () => {
  const historyAction = history.action;

  if (historyAction === HistoryAction.PUSH) {
    history.goBack();
  } else {
    history.push(AppRoute.ROOT);
  }
};

export default history;
