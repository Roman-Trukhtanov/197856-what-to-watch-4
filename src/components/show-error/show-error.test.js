import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import ShowError from "./show-error";
import history from "../../history";
// import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

describe(`ShowError component`, () => {
  it(`Render ShowError`, () => {
    const store = mockStore({});

    const tree = renderer.create(
        <Provider store={store}>
          <Router
            history={history}
          >
            <ShowError errorMessage={`404: Page not found...`}/>
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
