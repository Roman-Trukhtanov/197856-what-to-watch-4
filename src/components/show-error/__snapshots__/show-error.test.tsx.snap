// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowError component Render ShowError 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
  </header>
  <div
    className="sign-in user-page__content"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="sign-in__message"
    >
      <p>
        Something went wrong, please try again...
      </p>
    </div>
    <div
      className="sign-in__message"
    >
      <p>
        404: Page not found...
      </p>
    </div>
    <div
      className="sign-in__message"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "24px",
            "lineHeight": "26px",
          }
        }
      >
        Click to reload!
      </button>
    </div>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
